Examples/Idealgas.py

```diff
@@ -0,0 +1,150 @@
+from chiron.utils import PRNG, get_list_of_mass
+import jax.numpy as jnp
+
+# particles are non interacting
+cutoff = 0.0 * unit.nanometer
+ideal_gas_potential = IdealGasPotential(ideal_gas.topology)
+
+from chiron.states import SamplerState, ThermodynamicState
+
+# define the thermodynamic state
+thermodynamic_state = ThermodynamicState(
+    potential=ideal_gas_potential,
+    temperature=temperature,
+    pressure=pressure,
+)
+
+PRNG.set_seed(1234)
+
+
+# define the sampler state
+sampler_state = SamplerState(
+    x0=ideal_gas.positions,
+    current_PRNG_key=PRNG.get_random_key(),
+    box_vectors=ideal_gas.system.getDefaultPeriodicBoxVectors(),
+)
+
+from chiron.neighbors import PairList, OrthogonalPeriodicSpace
+
+# define the pair list for an orthogonal periodic space
+# since particles are non-interacting, this will not really do much
+# but will be used to appropriately wrap particles in space
+nbr_list = PairList(OrthogonalPeriodicSpace(), cutoff=cutoff)
+nbr_list.build_from_state(sampler_state)
+
+from chiron.reporters import MCReporter
+
+# initialize a reporter to save the simulation data
+filename = "test_mc_ideal_gas.h5"
+import os
+
+if os.path.isfile(filename):
+    os.remove(filename)
+reporter = MCReporter(filename, 100)
+
+
+from chiron.mcmc import (
+    MetropolisDisplacementMove,
+    MonteCarloBarostatMove,
+    MoveSchedule,
+    MCMCSampler,
+)
+
+# initialize the displacement move
+mc_barostat_move = MonteCarloBarostatMove(
+    volume_max_scale=0.2,
+    nr_of_moves=10,
+    reporter=reporter,
+    update_stepsize=True,
+    update_stepsize_frequency=100,
+)
+
+# initialize the barostat move and the move schedule
+metropolis_displacement_move = MetropolisDisplacementMove(
+    displacement_sigma=0.1 * unit.nanometer,
+    nr_of_moves=100,
+    update_stepsize=True,
+    update_stepsize_frequency=100,
+)
+
+# define the move schedule
+move_set = MoveSchedule(
+    [
+        ("MetropolisDisplacementMove", metropolis_displacement_move),
+        ("MonteCarloBarostatMove", mc_barostat_move),
+    ]
+)
+
+sampler = MCMCSampler(move_set)
+sampler.run(
+    sampler_state, thermodynamic_state, n_iterations=10, nbr_list=nbr_list
+)  # how many times to repeat
+
+# get the volume from the reporter
+volume = reporter.get_property("volume")
+step = reporter.get_property("elapsed_step")
+
+
+import matplotlib.pyplot as plt
+
+plt.plot(step, volume)
+plt.show()
+
+# get expectations
+ideal_volume = ideal_gas.get_volume_expectation(thermodynamic_state)
+ideal_volume_std = ideal_gas.get_volume_standard_deviation(thermodynamic_state)
+
+print("ideal volume and standard deviation: ", ideal_volume, ideal_volume_std)
+
+
+volume_mean = jnp.mean(jnp.array(volume)) * unit.nanometer**3
+volume_std = jnp.std(jnp.array(volume)) * unit.nanometer**3
+
+
+print("measured volume and standard deviation: ", volume_mean, volume_std)
+
+# get the masses of particles from the topology
+masses = get_list_of_mass(ideal_gas.topology)
```

Comment:
Yes, it's a bit messy.  The problem stems from using the openmm topology object as our topology. This was done initially for convenience as we were working on getting things set up and validating against openmmtools test systems.  I think when we refactor all the tests, we should also implement our own topology object that is more jax/numpy friendly (with the ability to easy convert from an openmm topology object).

Reply:
The use of helper functions for that purpose is appropriate, in my opinion. It is still possible to do `get_list_of_mass(top).sum()`.

But if we move to a `chiron` test system implementation, I agree that this can be solved as property.

Reply:
I create a new issue here  (#32 ) about the creation of our own topology object, rather than using openmm. This should be implement in conjunction with the unit change over and creation of our own testing suite.